pinot-query-planner/src/main/java/org/apache/pinot/query/planner/QueryPlan.java

```diff
@@ -39,13 +45,15 @@
 public class QueryPlan {
   private final List<Pair<Integer, String>> _queryResultFields;
   private final Map<Integer, StageNode> _queryStageMap;
-  private final Map<Integer, StageMetadata> _stageMetadataMap;
+  private final List<StageMetadata> _stageMetadataList;
+  private final Map<Integer, DispatchablePlanMetadata> _dispatchablePlanMetadataMap;
```

Comment:
These maps can be simplified into List (key is always 0 to `numStages - 1`). Do you see the need of keeping it as map long term wise?

Reply:
keeping it as a map will probably be changed going forward. currently this map is not used anywhere in wire protocol so we should be able to individually change this going forward. will follow up if we needed to change this in the batch-dispatcher support PR

Reply:
related to: https://github.com/apache/pinot/pull/10673#discussion_r1175377455 will clean up afterwards once we introduce per-server dispatch logic